test/adapters/signatures.test.js

```diff
@@ -0,0 +1,156 @@
+const arbitraryMsgHash = '0xec4870a1ebdcfbc1cc84b0f5a30aac48ed8f17973e0189abdb939502e1948238'
+const magicValue = '0x1626ba7e'
+
+function getProposalCounter() {
+  return proposalCounter().next().value;
+}
+
+describe.only("Adapter - Signatures", () => {
+  before("deploy dao", async () => {
+    const { dao, adapters, extensions } = await deployDefaultDao({
+      owner: myAccount,
+    });
+    this.dao = dao;
+    this.adapters = adapters;
+    this.extensions = extensions;
+    this.snapshotId = await takeChainSnapshot();
+  });
+
+  beforeEach(async () => {
+    await revertChainSnapshot(this.snapshotId);
+    this.snapshotId = await takeChainSnapshot();
+  });
+
+  it("should be possible to create a signature proposal and successfully query the erc1271 interface if it passes", async () => {
+    const erc1271 = this.extensions.erc1271;
+    const voting = this.adapters.voting;
+    const signatures = this.adapters.signatures;
+
+    let proposalId = getProposalCounter();
+
+    //submit a sig
+    await signatures.submitProposal(
+      this.dao.address,
+      proposalId,
+      arbitraryMsgHash,
+      arbitrarySignatureHash,
+      magicValue,
+      [],
+      {
+        from: myAccount,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    await voting.submitVote(this.dao.address, proposalId, 1, {
+      from: myAccount,
+      gasPrice: toBN("0"),
+    });
+
+    //should not be able to process before the voting period has ended
+    try {
+      await signatures.processProposal(this.dao.address, proposalId, {
+        from: myAccount,
+        gasPrice: toBN("0"),
+      });
+    } catch (err) {
+      expect(err.reason).equal("proposal needs to pass");
+    }
+
+    await advanceTime(10000);
+    await signatures.processProposal(this.dao.address, proposalId, {
+      from: myAccount,
+      gasPrice: toBN("0"),
+    });
+    // query erc1271 interface
+    await checkSignature(erc1271, arbitraryMsgHash, arbitrarySignature, magicValue);
+  });
+
+  it("should not be possible to get a valid signature if the proposal fails", async () => {
+    const voting = this.adapters.voting;
+    const signatures = this.adapters.signatures;
+
+    let proposalId = getProposalCounter();
+    
+    //submit a sig
+    await signatures.submitProposal(
+      this.dao.address,
+      proposalId,
+      arbitraryMsgHash,
+      arbitrarySignatureHash,
+      magicValue,
+      [],
+      {
+        from: myAccount,
+        gasPrice: toBN("0"),
+      }
+    );
+
+    //Member votes on the signature proposal
+    await voting.submitVote(this.dao.address, proposalId, 2, {
+      from: myAccount,
+      gasPrice: toBN("0"),
+    });
+
+    await advanceTime(10000);
+
+    try {
+      await signatures.processProposal(this.dao.address, proposalId, {
+        from: myAccount,
+        gasPrice: toBN("0"),
+      });
+    } catch (err) {
+      expect(err.reason).equal("proposal needs to pass");
```

Comment:
add a test to show that checkSignature does not pass here.

Reply:
done